qiskit_machine_learning/algorithms/distribution_learners/qgan/quantum_generator.py

```diff
@@ -417,10 +421,22 @@ def gradient_function(current_point):
                 quantum_instance, params=current_point, shots=self._shots
             )
             prediction_generated = discriminator.get_label(generated_data, detach=True)
-            op = ~CircuitStateFn(primitive=self.generator_circuit)
+            op = CircuitStateFn(primitive=self.generator_circuit)
             grad_object = gradient_object.convert(operator=op, params=free_params)
             value_dict = {free_params[i]: current_point[i] for i in range(len(free_params))}
-            analytical_gradients = np.array(grad_object.assign_parameters(value_dict).eval())
+            if quantum_instance.backend is not None:
```

Comment:
According to the hierarchy of calls, `quantum_instance` can be `None` here. Should it be a test like `if quantum_instance is not None:`?